src/components/social-table/SocialTable.jsx

```diff
@@ -5,12 +5,38 @@ import twitterImg from 'assets/image/svg/twitter.svg'
 import whatsappImg from 'assets/image/svg/whatsapp.svg'
 
 const socialLinks = [
-    {name: 'facebook', img: facebookImg, caption: 'Share on Facebook'},
-    {name: 'twitter', img: twitterImg, caption: 'Share on Twitter'},
-    {name: 'whatsapp', img: whatsappImg, caption: ''},
+    {
+        name: 'facebook',
+        img: facebookImg,
+        caption: 'Share on Facebook',
+        hrefShare (shareUrl) {
+            return `https://www.facebook.com/sharer/sharer.php?u=${shareUrl}`
+        },
```

Comment:
I don't like the idea of storing a function here. Main reason is that your function has the same name in all 3 objects, but the body of the function is different everywhere. if it was the same, then I could consider this implementation.

As an option I can suggest to do this. 
instead of a function just add a new attribute, e.g. `baseShareLink`. 
```
const fb = {name: 'facebook', img: facebookImg, baseShareLink: 'https://www.facebook.com/sharer/sharer.php?u='}
```
create a separate component for social links

```tsx
const SocialLink = (props) => {
  const href = props.baseShareLink ? `${props.baseShareLink}${currentUrl}` : ''
  return <a rel='noreferrer'
                                   target='_blank'
                                   className={style.socialTable__ref}
                                   href={href}>
                                    <img src={props.img} alt={props.name} width="20px" height="20px"/>
                                    <span className={style.socialTable__caption}>{'    ' + props.caption}</span>
                                </a>
}
```